File: apply_translation.py
#!/usr/bin/env python3
import json
import re

# 파일 읽기
with open('bundle.js', 'r', encoding='utf-8') as f:
    bundle_content = f.read()

with open('translations-final.json', 'r', encoding='utf-8') as f:
    translations = json.load(f)

# 번역 적용
result = bundle_content

# 번역 키를 길이 순으로 정렬 (긴 것부터 - 부분 매칭 방지)
sorted_translations = sorted(translations.items(), key=lambda x: len(x[0]), reverse=True)

for en, ko in sorted_translations:
    # JSON escape
    en_escaped = json.dumps(en)[1:-1]  # 따옴표 제거
    ko_escaped = json.dumps(ko)[1:-1]
    
    # 큰따옴표로 감싸진 경우
    result = result.replace(f'"{en_escaped}"', f'"{ko_escaped}"')
    # 작은따옴표로 감싸진 경우
    result = result.replace(f"'{en_escaped}'", f"'{ko_escaped}'")

# 결과 저장
with open('bundle-ko.js', 'w', encoding='utf-8') as f:
    f.write(result)

print('✅ Translation applied successfully!')
print(f'📝 Output: bundle-ko.js')
print(f'📊 Translations applied: {len(translations)}')

File: apply-translation.js
const fs = require('fs');
const path = require('path');

// 파일 경로
const bundlePath = path.join(__dirname, 'bundle.js');
const translationsPath = path.join(__dirname, 'translations-final.json');
const outputPath = path.join(__dirname, 'bundle-ko.js');

// 파일 읽기
const bundleContent = fs.readFileSync(bundlePath, 'utf8');
const translations = JSON.parse(fs.readFileSync(translationsPath, 'utf8'));

// 번역 적용
let result = bundleContent;

// JSON.stringify를 사용해서 정확한 문자열 매칭
Object.entries(translations).forEach(([en, ko]) => {
  // 문자열 리터럴로 escape된 형태로 교체
  const escapedEn = JSON.stringify(en).slice(1, -1); // 따옴표 제거
  const escapedKo = JSON.stringify(ko).slice(1, -1);
  
  // 다양한 패턴으로 교체
  // 1. 큰따옴표로 감싸진 경우
  result = result.split(`"${escapedEn}"`).join(`"${escapedKo}"`);
  // 2. 작은따옴표로 감싸진 경우
  result = result.split(`'${escapedEn}'`).join(`'${escapedKo}'`);
});

// 결과 저장
fs.writeFileSync(outputPath, result, 'utf8');

console.log('✅ Translation applied successfully!');
console.log(`📝 Output: ${outputPath}`);
console.log(`📊 Translations applied: ${Object.keys(translations).length}`);
